.github/workflows/main.yml

```diff
@@ -52,7 +52,10 @@ jobs:
             python setup.py sdist
       - name: Publish package to PyPI
         uses: pypa/gh-action-pypi-publish@master
+        # release dev versions on PR or stable only when tag on master
+        if: |
+          (github.event_name == 'release' && github.head_ref == 'master') ||
+          (github.event_name == 'pull_request' && github.head_ref != 'master')
```

Comment:
What is the reason to publish on pypi a PR before it's merged? A few reasons why it doesn't seem a good idea to me, to be discussed:
- Maybe the PR won't be merged, so it seems undesirable to create a dev package for that.
- How the dev version number is determined? If there are more than one PR open at the same time, could they conflict each other?
- If the PR can be created by anyone and a package is automatically published, it could cause a security threat. Only the maintainers of the project should be able to publish a package. Is this the case?